retworkx-core/src/connectivity/conn_components.rs

```diff
@@ -0,0 +1,98 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use hashbrown::HashSet;
+use std::collections::VecDeque;
+
+use petgraph::graph::{Graph, NodeIndex};
+use petgraph::visit::{VisitMap, Visitable};
+use petgraph::Undirected;
+
+/// Given an undirected graph, a start node and the visit_map for
+/// the graph, this function returns a connected component set.
+///
+/// :param Graph graph: The input graph to find the connected
+///     components for.
+/// :param NodeIndex start: The node to start from.
+/// :param Visitable::Map discovered: The visit map for the graph.
+///
+/// :return: A set of connected components for the start node
+/// :rtype: HashSet<usize>
+pub fn bfs_undirected(
```

Comment:
We should make the input graph type here generic over the graph traits the function requires. Instead of hard coded the type to be `Graph`.

Reply:
Done in https://github.com/Qiskit/retworkx/pull/595/commits/48eb720cc406775f7d59d39d001dfdab40d03161.